Automattic/woocommerce-payments
# Revemoved pre-fretch session for button

Fixes #7256

#### Changes proposed in this Pull Request

When we preload the checkout response to pass to WooPay, [it creates a draft order](https://github.com/woocommerce/woocommerce-blocks/blob/38ae651c9c3a19ae4ba32773d2d35b351cd081e7/src/StoreApi/Routes/V1/Checkout.php#L150C22-L150C22). Ever since the session data prefetch was implemented in #7199, this preload happens whenever the WooPay button is displayed. Because of this, a draft order is created just by going to the checkout page. This reserves the stock, but the checkout tries to create a new one, and it fails because there's no stock left.
This PR removes this prefetch, making it work.
It **doesn't fix** the same error after the modal is open and then closed, or redirected to WooPay and then getting back. These issues belong to #7257.

#### Testing instructions

1. Enable the WooPay button on the cart and checkout pages
1. Create an item with only 1 in stock and run a few test scenarios (you have to update the stock back to 1 after each test):
    1. Using the WooPay button from the cart page and paying with WooPay.
    1. Going to the cart page before going to the checkout page and paying with regular checkout (make sure not to type a valid woopay address in the field).
    1. Going to the cart page before going to the checkout page and paying with WooPay Button.
    1. Going to the cart page before going to the checkout page and paying with WooPay email input.
    1. Going directly to the checkout page and paying with regular checkout (make sure not to type a valid woopay address in the field).
    1. Going directly to the checkout page and paying with WooPay Button.
    1. Going directly to the checkout page and paying with WooPay email input.



-------------------


**Post merge**
